android-components/components/feature/top-sites/src/main/java/mozilla/components/feature/top/sites/DefaultTopSitesStorage.kt

```diff
@@ -105,14 +107,17 @@ class DefaultTopSitesStorage(
                     .filter { providerConfig.providerFilter?.invoke(it) ?: true }
```

Comment:
We are only filtering the top sites provded by Contile. So, we should just filter here instead of filtering the providerTopSites + pinnedSites below.

Reply:
Yes, we would like to filter the default or pinned top sites. This is because we add some default top sites, let's say [A, B, C]; while during shopping fes, we may still need B but with a different url (for example a diffrenet reference code or something else), let's call it B' . So we put B' into the contile provider. At the same time we don't wanna users see two different Bs in their top sites, so we want to hide the original default top site B, to eventually show users [B', A, C]. 
For users who don't have contile service on, they'll just keep [A, B, C] without showing B' or hiding B. 
The reason we need this more complicated method to deal with these top sites is that we used to add our collaborations all into default top sites in MozillaOnline builds before contile service available.

Reply:
@rxumoz One more clarifying question from me as I continue to think more about the this. Is MozillaOnline also using the same service that the Contile team built? For example, I am wondering if the existing caching and fallback strategy that the Contile team uses are also relevant for MozillaOnline. Currently, we are piggybacking on the existing `ContileTopSitesProvider` implementation which was designed for the Contile sponsored tiles. However, if one of the services changes in the future because they weren't using the same implementation and becomes incompatible for eachother, it does raise a question whether or not we should introduce a separate `MozillaOnlineTopSitesProvider` and `MozillaOnlineTopSitesStorage` that your team could fine tune for your own use case if the services are indeed different. We would duplicate the class logic that makes sense and modify it to your own needs.

Reply:
Our main aim is to show some sponsored top sites defined by server in MozillaOnline builds, which I believe is the same as contiles. As for caching and fallback, we do not have any special requirement. From this perspective I think current `ContileTopSitesProvider` is good for us.
For exclusions, due to historical reasons (we put sponsored sites into default top sites), we have to deal with old default top sites to avoid duplication. I chose to use the same server to do this for agile control, but it doesn't have to be this way if there is any better method. But no matter what method we use, I think we still need to handle it within `DefaultTopSitesStorage` since this is the place we used to put sponsored sites in.

Reply:
@rxumoz I am just checking with the contile team to confirm the server is the same as theirs. I don't really have any concerns if they come back to confirm that everything is pretty much the same as their team. My remaining ask then would be to fix the bug pointed out in https://github.com/mozilla-mobile/firefox-android/pull/898#discussion_r1203370159 so that you will get the right number of top sites after filtering.

Reply:
I think currently our server does not have any config for this, just return a JSON, that's all. BTW I change the endpoint url to https://download-ssl.firefox.com.cn/fenix-contile/d3.json  , since the original one (https://download-ssl.firefox.com.cn/test20220920/d3.json) is for test.